examples/groth16/rust/src/main.rs

```diff
@@ -0,0 +1,56 @@
+use ark_bn254::{Bn254, Fr};
+use ark_circom::CircomBuilder;
+use ark_circom::CircomConfig;
+use ark_groth16::{prepare_verifying_key, Groth16};
+use ark_serialize::CanonicalSerialize;
+use ark_snark::SNARK;
+
+#[tokio::main]
+async fn main() {
+    // Load the WASM and R1CS for witness and proof generation
+    let cfg =
+        CircomConfig::<Fr>::new("../circuit/main_js/main.wasm", "../circuit/main.r1cs").unwrap();
+    let mut builder = CircomBuilder::new(cfg);
+
+    // Private inputs: A factorisation of a number
```

Comment:
In case you think it's a relevant fun fact to write here, this number is the 5th Fermat Number 2^32 + 1.